### PR TITLE
Migrating 'html_safe' strings to use the markdown version instead

### PR DESCRIPTION
**Pull request by daynew**
# Description
We had translated strings which had raw HTML in them. We have created a "_markdown" version of these strings and this PR is where we use them. All the updates are around the sign-in screens or instances where we accept the TOS. There are also some styling changes in order to make things more consistent between the sign in pages.
* Changed markdown renderer options so that `/` and `//` URLs can be rendered correctly.
* Used new translated "_markdown" strings on the old/new sign-in pages and the homepage TOS pop-up.
* Updated the "old" sign-in page to use `<label>`s so that it was similar to the "new" sign-in page and also to improve accessibility.
* Tweaked spacing slightly to make things more even and consistent.

### _old_sign_up_form
_old_sign_up_form
### _new_sign_up_form
_new_sign_up_form
### _finish_sign_up
Note* you can ignore the "****". It renders that way because I did a hack to render this page and I forgot to fill in the user e-mail before taking a screenshot. I have tested that it correctly displays the e-mail.
_finish_sign_up
### accept_terms_reminder
accept_terms_reminder

## Links
- [jira](https://codedotorg.atlassian.net/browse/FND-818)

## Testing story
* `./bin/dashboard-server`
  Manually tested using (Linux) X (Chrome, Firefox)

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/views/devise/registrations/_old_sign_up_form.html.haml [hunk 1]
```diff
@@ -117,16 +117,17 @@
       %br
       #terms_of_service_version-block.itemblock
         .error_in_field.checkbox-error
-        %br
-        = f.label :terms_of_service_version do
+        = f.label :terms_of_service_version, {class: 'terms'} do
```
Comment (Hamms) [inline dashboard/app/views/devise/registrations/_old_sign_up_form.html.haml:121]: note that labels are [inline elements that should only contain phrasing content](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label#Technical_summary)

Rather than trying to manipulate this label with CSS to support including the kind of content we want, I would suggest extracting the content that should not be contained in a label out to a sibling element, and updating the label to only contain label-like content.

ie, instead of something like

```html
<label>
  <input class="terms-checkbox" type="checkbox" value="1" name="user[terms_of_service_version]" id="user_terms_of_service_version">
  <div class="teacher-options" style="display: none">
    <p>I agree to the <a href="//code.org/tos" target="_blank">Terms of Service</a> and <a href="//code.org/privacy" target="_blank">Privacy Policy</a>.</p>
  </div>
  <div class="student-options">
    <p>I agree to the <a href="//code.org/tos" target="_blank">Terms of Service</a> and <a href="//code.org/privacy" target="_blank">Privacy Policy</a>.</p>
    <p>If I am under 13 years of age, I confirm that I have my parent or legal guardian's permission to use the Code.org services.</p>
  </div>
</label>
```

do something like

```html
<div>
  <div class="teacher-options" style="display: none">
    <p>I agree to the <a href="//code.org/tos" target="_blank">Terms of Service</a> and <a href="//code.org/privacy" target="_blank">Privacy Policy</a>.</p>
  </div>
  <div class="student-options">
    <p>I agree to the <a href="//code.org/tos" target="_blank">Terms of Service</a> and <a href="//code.org/privacy" target="_blank">Privacy Policy</a>.</p>
    <p>If I am under 13 years of age, I confirm that I have my parent or legal guardian's permission to use the Code.org services.</p>
  </div>
  <label>
    <input class="terms-checkbox" type="checkbox" value="1" name="user[terms_of_service_version]" id="user_terms_of_service_version">
    I agree
  </label>
</div>
```

File: lib/cdo/i18n_backend.rb [hunk 2]
```diff
@@ -64,7 +64,9 @@ module MarkdownTranslate
       def translate(locale, key, options = ::I18n::EMPTY_HASH)
         translation = super(locale, key, options)
         if options.fetch(:markdown, false)
-          @renderer ||= Redcarpet::Markdown.new(Redcarpet::Render::Safe)
+          # The safe_links_only just makes sure that the URL is a "safe" web one which starts with: "#", "/", "http://", "https://", "ftp://", "mailto:"
```
Comment (Hamms) [inline lib/cdo/i18n_backend.rb]: what are the "unsafe" links that this setting is attempting to protect us from?

File: dashboard/app/views/devise/registrations/_old_sign_up_form.html.haml (hunk 1 above)
Comment (daynew) [reply dashboard/app/views/devise/registrations/_old_sign_up_form.html.haml:121]: Done. Please checkout the new screenshots to preview what it looks like.

File: lib/cdo/i18n_backend.rb (hunk 2 above)
Comment (daynew) [reply lib/cdo/i18n_backend.rb]: I put my investigation notes in the comments. Let me know if that is good enough.

File: dashboard/app/assets/stylesheets/application.scss [hunk 3]
```diff
@@ -1580,6 +1580,10 @@ a.download-video {
     width: 800px;
   }
 
+  .finish-signup #instructions p {
+    display: inline;
```
Comment (Hamms) [inline dashboard/app/assets/stylesheets/application.scss:1584]: why is this necessary? I don't think it's desirable to make paragraphs behave as if they aren't paragraphs.

File: dashboard/app/assets/stylesheets/application.scss [hunk 4]
```diff
@@ -1730,7 +1730,16 @@ a.download-video {
     margin-bottom: 5px;
     margin-top: 5px;
   }
-  .terms-checkbox, .race-checkbox {
+
+  .terms-container {
+    width: 600px;
+  }
+  .terms-text p {
+    display: inline;
```
Comment (Hamms) [inline dashboard/app/assets/stylesheets/application.scss:1743]: also here

**Review by Hamms — COMMENTED** [review]
comments about safe links look great, the refactoring of the signup page looks great!

We should probably get signoff from someone on the user-visible changes to the signup page, but I'm not sure who; Bryan would probably know.

One remaining question about inline paragraphs, but otherwise this is looking great.

**Comment by dju90** [comment]
Tagging involved parties:
Discussed the changes in the "accept ToS and PP" checkbox with @boatmarina and it is much preferable to keep the checkbox to the left of the text, despite @Hamms' point that it's against html specifications to have a paragraph tag inside a form label. This conversation may eventually become moot, as (pending consultation with our lawyer) we hope to get rid of the checkbox and just have "By clicking 'Go to my account', you agree...", which is a common pattern across the web. Because of this, I would like to keep the text in the checkbox label for now.

**Comment by daynew** [comment]
Created https://codedotorg.atlassian.net/browse/FND-893 for fixing up our markup content usage which violates the HTML spec. I also added TODO's to fix the usage next sprint.

File: dashboard/app/assets/stylesheets/application.scss (hunk 3 above)
Comment (daynew) [reply dashboard/app/assets/stylesheets/application.scss:1584]: I created a JIRA task and TODO's to resolve this next sprint

File: dashboard/app/assets/stylesheets/application.scss (hunk 4 above)
Comment (daynew) [reply dashboard/app/assets/stylesheets/application.scss:1743]: I created a JIRA task and TODO's to resolve this next sprint

File: dashboard/app/assets/stylesheets/application.scss [hunk 5]
```diff
@@ -1580,6 +1580,10 @@ a.download-video {
     width: 800px;
   }
 
+  .finish-signup #instructions p {
```
Comment (Hamms) [inline dashboard/app/assets/stylesheets/application.scss:1583]:
```suggestion
  /* FND-893 - Stop using markdown in <Label>s*/
  .finish-signup #instructions p {
```

File: dashboard/app/views/zendesk_session/index.html.haml [hunk 6]
```diff
@@ -1 +1 @@
-%p!= t('zendesk_too_young_message')
+%p!= t('zendesk_too_young_message_markdown', markdown: true)
```
Comment (Hamms) [inline dashboard/app/views/zendesk_session/index.html.haml]:
```suggestion
= t('zendesk_too_young_message_markdown', markdown: true).html_safe
```

File: dashboard/app/views/devise/registrations/_sign_up.html.haml [hunk 7]
```diff
@@ -5,9 +5,9 @@
     .span9
       %h1= t('signup_form.title')
       - if @already_hoc_registered
-        %p= t('signup_form.hoc_already_signed_up_content_post_hoc', studio_url: CDO.studio_url('/')).html_safe
+        %p= t('signup_form.hoc_already_signed_up_content_post_hoc_markdown', studio_url: CDO.studio_url('/'), markdown: true).html_safe
```
Comment (Hamms) [inline dashboard/app/views/devise/registrations/_sign_up.html.haml]:
```suggestion
        = t('signup_form.hoc_already_signed_up_content_post_hoc_markdown', studio_url: CDO.studio_url('/'), markdown: true).html_safe
```

File: dashboard/app/views/devise/registrations/_sign_up.html.haml [hunk 8]
```diff
@@ -5,9 +5,9 @@
...
       - else
-        %p= t('signup_form.overview').html_safe
+        %p= t('signup_form.overview_markdown', markdown: true).html_safe
```
Comment (Hamms) [inline dashboard/app/views/devise/registrations/_sign_up.html.haml]:
```suggestion
        = t('signup_form.overview_markdown', markdown: true).html_safe
```

File: dashboard/app/views/devise/registrations/_sign_up.html.haml [hunk 9]
```diff
@@ -5,9 +5,9 @@
     .span9
       %h1= t('signup_form.title')
```
Comment (Hamms) [inline dashboard/app/views/devise/registrations/_sign_up.html.haml:7]:
```suggestion
      -# FND-893 - Stop using markdown inside headers
      %h1= t('signup_form.title')
```

**Review by Hamms — APPROVED** [review]
couple nits, otherwise LGTM!
